src/index.js

```diff
@@ -132,148 +125,93 @@ export default class ToggleBlock {
+      const { type, data } = block;
 
-    if (validItems || savedData.text.trim()) {
-      return true;
-    }
-    return false;
-  }
+      this.api.blocks.insert(type, data, {}, index += 1, true);
 
-  /**
-   * First saves the current toggle status. After, calls the method to insert
-   * a paragraph, as this leaves the toggle open, asks if the toggle status
-   * before calling the method is different from the current status, if it's,
-   * it executes some instructions to collapse the toggle and return its
-   * original status.
-   *
-   * @param {string} paragraph - paragraph text
-   */
-  _renderParagraph(paragraph = '') {
-    const currenStatus = this.data.status;
+      const newBlock = this.api.blocks.getBlockByIndex(index);
+      const { holder } = newBlock;
+      const content = holder.firstChild;
+      const item = content.firstChild;
 
-    this.insertParagraph(paragraph);
+      holder.addEventListener('keydown', this.createParagraphFromIt.bind(this));
+      holder.setAttribute('foreignKey', foreignKey);
+      holder.setAttribute('id', crypto.randomUUID());
 
-    if (currenStatus !== this.data.status) {
-      this.wrapper.firstChild.innerHTML = this._resolveToggleAction();
-      this._hideAndShowParagraphs();
-    }
-  }
+      item.classList.add('toggle-block__item');
+    });
 
-  /**
-   * First checks the status of a toggle, if this is 'closed' then open it.
-   * After, calls the method to create a paragraph and the result is inserted
-   * as a toggle child.
-   *
-   * @param {string} text - paragraph text
-   */
-  insertParagraph(text = '') {
-    if (this.data.status === 'closed') {
-      this.wrapper.firstChild.innerHTML = this._resolveToggleAction();
-      this._hideAndShowParagraphs();
+    if (editorBlocks > 1) {
+      this._hideAndShowBlocks(originalIndex, this.data.items.length);
+    } else {
+      this._hideAndShowBlocks(originalIndex - 1, this.data.items.length);
     }
-
-    const paragraph = this.createParagraph(text);
-
-    this.wrapper.appendChild(paragraph);
-  }
-
-  /**
-   * * Creates a paragraph view
-   * @param {string} content - paragraph text
-   * @returns {HTMLDivElement}
-   */
-  createParagraph(content = '') {
-    const newParagraph = document.createElement('div');
-
-    newParagraph.classList.add('toggle-block__paragraph');
-    newParagraph.setAttribute('id', crypto.randomUUID());
-    newParagraph.addEventListener('keyup', this.removeParagraphFromIt.bind(this, newParagraph.id));
-    newParagraph.addEventListener('keydown', this.createParagraphFromIt.bind(this, newParagraph.id));
-    newParagraph.contentEditable = true;
-    newParagraph.innerHTML = content || '';
-
-    return newParagraph;
   }
 
   /**
    * Converts the toggle status to its opposite, including its icon.
    * If the toggle status is open, then now will be closed and its icon
    * will be the main. Otherwise, will be open and its icon will be the
-   * secundary.
+   * Secondary.
```

Comment:
@juliansg022 A little detail, here you should change the method description.

Reply:
@MarioRodriguezS You're right, I missed it.